moveit_ros/moveit_servo/test/servo_launch_test_common.hpp

```diff
@@ -113,26 +113,28 @@ class ServoFixture : public ::testing::Test
   {
     // Start client
     client_servo_start_ = node_->create_client<std_srvs::srv::Trigger>("/start_servo");
-    while (!client_servo_start_->wait_for_service(1s))
+    while (!client_servo_start_->service_is_ready())
```

Comment:
Check out wait_for_service() defined here:

https://github.com/ros2/rclcpp/blob/bbf2c4cbfb006245c784f39ee8f8f9e56431c91e/rclcpp/include/rclcpp/parameter_client.hpp#L226

It calls wait_for_service_nanoseconds() ... which again calls wait_for_service(). Do you think that recursion is the root cause? In other words, the source of the infinite loop is in rclcpp?

Here's the definition of wait_for_service_nanoseconds():

https://github.com/ros2/rclcpp/blob/633e1157f8aeee9249afddcd0730c3f9bb894fce/rclcpp/src/rclcpp/parameter_client.cpp#L285

Reply:
Yes, I think what happens there would be fine if the cpu had enough cores to allow you to lock up one doing that.  It never sleeps in there so it may not ever give the cpu the chance to run servo so it can finish initalizing.